pkg/k8s/authz_test.go

```diff
@@ -58,3 +59,37 @@ subjects:
 		})
 	}
 }
+
+func TestServiceProfilesAccess(t *testing.T) {
+	fakeResources := []string{`
+kind: APIResourceList
+apiVersion: v1
+groupVersion: linkerd.io/v1alpha1
+resources:
+- name: serviceprofiles
+  singularName: serviceprofile
+  namespaced: true
+  kind: ServiceProfile
+  verbs:
+  - delete
+  - deletecollection
+  - get
+  - list
+  - patch
+  - create
+  - update
+  - watch
+  shortNames:
+  - sp`}
+
+	api, err := NewFakeAPI(fakeResources...)
+	if err != nil {
+		t.Fatalf("NewFakeAPI error: %s", err)
+	}
+
+	err = ServiceProfilesAccess(api)
+	// RBAC SSAR request failed, but the Discovery lookup succeeded
+	if !reflect.DeepEqual(err, errors.New("not authorized to access serviceprofiles.linkerd.io")) {
+		t.Fatalf("unexpected error: %s", err)
+	}
+}
```

Comment:
For my own understanding, how would one manage testing a positive result? I guess we'd have to add into `fakeResources` the RBAC to access serviceprofiles like we have in `controller-rbac.yaml`, but then how's the user of `NewFakeAPI` bound to those privileges?

Reply:
Right, that's the general idea. I haven't yet determined which mock resources we need enable a successful `SelfSubjectAccessReview` request.

This applies to every test in this file. I referenced the issue in a TODO when I introduced `authz_test.go`:
https://github.com/linkerd/linkerd2/blob/00a94be07375eb24ef9a15594fd9bc3518a221b1/pkg/k8s/authz_test.go#L15